### PR TITLE
Add `snapcraft list-keys`

### PR DESCRIPTION
**Pull request by cjwatson**
This augments `snap keys` by also showing which keys are registered to
your account on the store.

LP: #1621895

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by sergiusens** [comment]
I am going to surface this here just in case
```
======================================================================
FAIL: test_get_account_information_successfully (snapcraft.tests.store.test_store_client.GetAccountInformationTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/snapcore/snapcraft/snapcraft/tests/store/test_store_client.py", line 170, in test_get_account_information_successfully
    {'account_id': 'abcd'}, self.client.get_account_information())
AssertionError: {'account_id': 'abcd'} != {'account_id': 'abcd', 'account_keys': []}
- {'account_id': 'abcd'}
+ {'account_id': 'abcd', 'account_keys': []}
```

File: snapcraft/_store.py [hunk 1]
```diff
@@ -80,6 +81,16 @@ def logout():
     logger.info('Credentials cleared.')
 
 
+@contextmanager
```
Comment (sergiusens) [inline snapcraft/_store.py]: \o/

**Comment by cjwatson** [comment]
Whoops, sorry about that test failure.  Should be fixed now.

File: snapcraft/tests/test_commands_list_keys.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+)
+from snapcraft.tests.test_commands_register_key import (
+    get_sample_key,
+    mock_snap_output,
+)
+
+
+class ListKeysTestCase(tests.TestCase):
+
+    scenarios = [
```
Comment (josepht) [inline snapcraft/tests/test_commands_list_keys.py]: Is this used somewhere under the hood?  I don't see it referenced anywhere else.

File: snapcraft/tests/test_commands_list_keys.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            'account_keys': [
+                {
+                    'name': 'default',
+                    'public-key-sha3-384': (
+                        get_sample_key('default')['sha3-384']),
+                },
+            ],
+        }
+
+        main([self.command_name])
```
Comment (josepht) [inline snapcraft/tests/test_commands_list_keys.py]: Ah! I guess it's here that scenarios are used.  Nevermind above then, unless I'm missing something.

**Comment by josepht** [comment]
Looks good to me.  I agree with @sergiusens +1 on the contextmanager.

**Comment by sergiusens** [comment]
ok to test

**Pull request merged**